plugins/extractors/service_yaml/service.go

```diff
@@ -0,0 +1,84 @@
+package serviceyaml
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/mitchellh/mapstructure"
+)
+
+type Service struct {
+	Name string `json:"name" yaml:"name" validate:"required"`
+	ID   string `json:"id" yaml:"id" validate:"required"`
+	Team struct {
+		ID    string `json:"id" yaml:"id"`
+		Name  string `json:"name" yaml:"name"`
+		Email string `json:"email" yaml:"email"`
+	} `json:"team" yaml:"team"`
+	Description string            `json:"description" yaml:"description"`
+	URL         string            `json:"url" yaml:"url" validate:"omitempty,url"`
+	Version     string            `json:"version" yaml:"version"`
+	Inputs      []Dataset         `json:"inputs" yaml:"inputs" validate:"dive"`
+	Outputs     []Dataset         `json:"outputs" yaml:"outputs" validate:"dive"`
+	CreateTime  time.Time         `json:"create_time" yaml:"create_time"`
+	UpdateTime  time.Time         `json:"update_time" yaml:"update_time"`
+	Labels      map[string]string `json:"labels" yaml:"labels"`
+}
+
+type Dataset struct {
+	Type       string                 `json:"type" yaml:"type" validate:"required,oneof=stream batch"`
+	Platform   string                 `json:"platform" yaml:"platform" validate:"required,oneof=kafka bigquery"`
```

Comment:
should we limit this field to only kafka and bigquery? or is it better if we leave it open?
`kafka` `bigquery` seems to be our setup only.

Also if we are talking about input, CMIIW it can come from other services as well. (worker hitting service's APIs)

Reply:
Also, let's call this extractor `application`.

Reply:
> or is it better if we leave it open?

How should we handle building lineage in that case? It might be surprising if only a specific subset is supported for building lineage. We could expand the list of supported platforms in this PR as well if you have something specific in mind.

> Also, let's call this extractor `application`.

@ravisuhag could you elaborate a bit on why? An `application` extractor that reads a file to emit a `Service` asset doesn't seem intuitive. Or are you suggesting we rename `{service => application}` in all places?

One more thing I would like to call out is that it is common for service configuration files to be called `application.yml` so that might be a little confusing to also have `application.meteor.yaml`.

Reply:
Yes, we can rename it to application everywhere. Since service is a concept we are already using. In terms of the name of the file, I think we can leave it to the user and the extractor can simply accept the file path.

Reply:
> Since service is a concept we are already using.

Could you help me understand this? Wasn't able to follow.

Reply:
https://github.com/odpf/proton/blob/23148b88c221d01a41aaacdff2e6700fc2d7ef25/odpf/assets/v1beta2/asset.proto#L25

Reply:
https://github.com/odpf/proton/pull/214

Reply:
> How should we handle building lineage in that case? It might be surprising if only a specific subset is supported for building lineage.

Hmm yeah that is actually true 🤔 , let's keep it as it is then for now